libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1774,10 +1779,11 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 		if (member == "selector")
 		{
 			auto const& functionType = dynamic_cast<FunctionType const&>(*_memberAccess.expression().annotation().type);
+			solAssert(!(functionType.kind() == FunctionType::Kind::Event));
```

Comment:
Please add a comment: All events should already be caught by the shortcut above.